src/thunderbots/software/test/grsim_backend/motion_controller.cpp

```diff
@@ -0,0 +1,659 @@
+{
+    Robot robot             = Robot(4, Point(0, 0), Vector(0.0, 0.0), Angle::ofRadians(0),
+                        AngularVelocity::ofRadians(0), current_time);
+    const double delta_time = TIME_STEP;
+    Point destination       = Point(0., 0);
+    Angle destination_angle = Angle::ofDegrees(90);
+    double destination_speed = 0;
+    int iteration_count;
+    MotionController::Velocity robot_velocities;
+    Angle new_orientation;
+    AngularVelocity final_angular_speed = AngularVelocity::ofRadians(0);
+
+    for (iteration_count = 0; iteration_count < TOTAL_STEPS; iteration_count++)
+    {
+        robot_velocities = MotionController::bangBangVelocityController(
+            robot, destination, destination_speed, destination_angle, delta_time);
+
+        new_orientation =
+            robot.orientation() +
+            Angle::ofRadians(robot.angularVelocity().toRadians() * delta_time);
+
+        Robot tempRobo =
+            Robot(4, robot.position(), robot_velocities.linear_velocity, new_orientation,
+                  robot_velocities.angular_velocity, current_time);
+        robot.updateState(robot.position(), robot_velocities.linear_velocity,
+                          new_orientation, robot_velocities.angular_velocity,
+                          current_time);
+    }
+
+    EXPECT_NEAR(robot.orientation().toRadians(), destination_angle.toRadians(),
+                POSITION_TOLERANCE);
+    EXPECT_NEAR(robot.angularVelocity().toRadians(), final_angular_speed.toRadians(),
+                POSITION_TOLERANCE);
+}
+
+TEST_F(MotionControllerTest, negative_rotation_position_test)
+{
+    Robot robot             = Robot(4, Point(0, 0), Vector(0.0, 0.0), Angle::ofRadians(0),
+                        AngularVelocity::ofRadians(0), current_time);
+    const double delta_time = TIME_STEP;
+    Point destination       = Point(0., 0);
+    Angle destination_angle = Angle::ofRadians(-1);
+    double destination_speed = 0;
+    int iteration_count;
+    MotionController::Velocity robot_velocities;
+    Angle new_orientation;
+    AngularVelocity final_angular_speed = AngularVelocity::ofRadians(0);
+
+    for (iteration_count = 0; iteration_count < TOTAL_STEPS; iteration_count++)
+    {
+        robot_velocities = MotionController::bangBangVelocityController(
+            robot, destination, destination_speed, destination_angle, delta_time);
+
+        new_orientation =
+            robot.orientation() +
+            Angle::ofRadians(robot.angularVelocity().toRadians() * delta_time);
+
+        robot.updateState(robot.position(), robot_velocities.linear_velocity,
+                          new_orientation, robot_velocities.angular_velocity,
+                          current_time);
+    }
+
+    EXPECT_NEAR(robot.orientation().toRadians(), destination_angle.toRadians(),
+                POSITION_TOLERANCE);
+    EXPECT_NEAR(robot.angularVelocity().toRadians(), final_angular_speed.toRadians(),
+                POSITION_TOLERANCE);
+}
+
```

Comment:
Is this being addressed in a future PR?

Reply:
So this PR  was starting to block stuff, so I somewhat let this one slide (as it shouldn't change the correctness of the software here). Unfortunately given our priorities and timeline right now I'd deem it unlikely this will be addressed.